source/validate_id.cpp

```diff
@@ -1710,10 +1710,12 @@ bool idUsage::isValid<SpvOpReturnValue>(const spv_instruction_t* inst,
                      << value.second.type_id << "' is missing or void.";
     return false;
   }
-  if (SpvOpTypePointer == valueType.second.opcode) {
+  if (addressingModel == SpvAddressingModelLogical &&
+      SpvOpTypePointer == valueType.second.opcode) {
     DIAG(valueIndex) << "OpReturnValue value's type <id> '"
                      << value.second.type_id
-                     << "' is a pointer, but a pointer can only be an operand "
+                     << "' is a pointer, but in the Logical addressing model "
+                        "a pointer can only be an operand "
                         "to OpLoad, OpStore, OpAccessChain, or "
                         "OpInBoundsAccessChain.";
```

Comment:
This is no longer accurate: https://www.khronos.org/registry/spir-v/specs/1.1/SPIRV.html#_universal_validation_rules